kouncil-backend/src/test/java/com/consdata/kouncil/serde/NoSchemaDeserializationServiceTest.java

```diff
@@ -0,0 +1,101 @@
+    @MockBean
+    private SchemaAwareClusterService schemaAwareClusterService;
+    @Autowired
+    private DeserializationService deserializationService;
+
+    @Test
+    public void should_deserialize_without_schema() {
+        // given
+        when(schemaAwareClusterService.clusterHasSchemaRegistry(anyString())).thenReturn(false);
+        ConsumerRecord<Bytes, Bytes> message = prepareConsumerRecord(
+                new Bytes(LOREM.getBytes(StandardCharsets.UTF_8)),
+                new Bytes(IPSUM.getBytes(StandardCharsets.UTF_8))
+        );
+
+        // when
+        DeserializedMessage deserializedMessage = deserializationService.deserialize("clusterId", message);
+
+        // then
+        assertThat(deserializedMessage.getKeyData().getDeserialized()).isEqualTo(LOREM);
+        assertThat(deserializedMessage.getKeyData().getMessageFormat()).isEqualTo(MessageFormat.STRING);
+        assertThat(deserializedMessage.getValueData().getDeserialized()).isEqualTo(IPSUM);
+        assertThat(deserializedMessage.getValueData().getMessageFormat()).isEqualTo(MessageFormat.STRING);
+    }
+
+    @Test
+    public void should_deserialize_without_schema_key_null() {
+        // given
+        when(schemaAwareClusterService.clusterHasSchemaRegistry(anyString())).thenReturn(false);
+        ConsumerRecord<Bytes, Bytes> message = prepareConsumerRecord(
+                null,
+                new Bytes(IPSUM.getBytes(StandardCharsets.UTF_8))
+        );
+
+        // when
+        DeserializedMessage deserializedMessage = deserializationService.deserialize("clusterId", message);
+
+        // then
+        assertThat(deserializedMessage.getKeyData().getDeserialized()).isNull();
+        assertThat(deserializedMessage.getKeyData().getMessageFormat()).isNull();
+        assertThat(deserializedMessage.getValueData().getDeserialized()).isEqualTo(IPSUM);
+        assertThat(deserializedMessage.getValueData().getMessageFormat()).isEqualTo(MessageFormat.STRING);
+    }
+
+    @Test
+    public void should_deserialize_without_schema_value_null() {
+        // given
+        when(schemaAwareClusterService.clusterHasSchemaRegistry(anyString())).thenReturn(false);
+        ConsumerRecord<Bytes, Bytes> message = prepareConsumerRecord(
+                new Bytes(LOREM.getBytes(StandardCharsets.UTF_8)),
+                null
+        );
+
+        // when
+        DeserializedMessage deserializedMessage = deserializationService.deserialize("clusterId", message);
+
+        // then
+        assertThat(deserializedMessage.getKeyData().getDeserialized()).isEqualTo(LOREM);
+        assertThat(deserializedMessage.getKeyData().getMessageFormat()).isEqualTo(MessageFormat.STRING);
+        assertThat(deserializedMessage.getValueData().getDeserialized()).isNull();
+        assertThat(deserializedMessage.getValueData().getMessageFormat()).isNull();
+    }
+
+    private ConsumerRecord<Bytes, Bytes> prepareConsumerRecord(Bytes key, Bytes value) {
```

Comment:
Może jescze to do jakiegoś utila? Czy to już przesada, co myślisz?

Reply:
też mi to się rzuciło i początkowo chciałem wydzielić, ale to tak naprawdę tylko tworzenie obiektu, nie wiem czy ma sens robić oddzielny util pod to, chociaż z drugiej strony sporo linii kodu zajmuje xd, hm

Reply:
zrobiłem i jakoś średnio mi podoba xd, więc chyba zostawię jak jest obecnie bez wydzielenia tego

Reply:
Ok